### PR TITLE
Add missing configuration for Bluetooth and storage partition for settings

### PR DESCRIPTION
**Pull request by ppryga-nordic**
There is missing default Kconfig configuration related with Bluetooth protocol for nRF54h20 DK.
There is missing storage partition configuration for settings subsystem for nRF54h20 DK.
There is missing nRF54H20 DK configuration in hci_ipc Bluetooth sample.

Add missing items.

### CODE REVIEW

**Comment by ppryga-nordic** [comment]
@hubertmis  please check the commit with storage partition if that fits into planned memory map.

File: boards/nordic/nrf54h20dk/Kconfig.nrf54h20dk [hunk 1]
```diff
@@ -6,3 +6,18 @@ config BOARD_NRF54H20DK
 	select SOC_NRF54H20_CPURAD if BOARD_NRF54H20DK_NRF54H20_CPURAD
 	select SOC_NRF54H20_CPUPPR if BOARD_NRF54H20DK_NRF54H20_CPUPPR || \
 				      BOARD_NRF54H20DK_NRF54H20_CPUPPR_XIP
+
+if BOARD_NRF54H20DK_NRF54H20_CPUAPP
+
+choice BT_HCI_BUS_TYPE
+	default BT_HCI_IPC if BT
```
Comment (57300) [inline boards/nordic/nrf54h20dk/Kconfig.nrf54h20dk:13]: Please create `Kconfig.defconfig` and move these changes there.

File: boards/nordic/nrf54h20dk/nrf54h20dk_nrf54h20-memory_map.dtsi [hunk 2]
```diff
@@ -174,11 +174,26 @@
 		#size-cells = <1>;
 
 		cpuapp_slot0_partition: partition@a6000 {
-			reg = <0xa6000 DT_SIZE_K(512)>;
+			reg = <0xa6000 DT_SIZE_K(488)>;
```
Comment (57300) [inline boards/nordic/nrf54h20dk/nrf54h20dk_nrf54h20-memory_map.dtsi:177]: If this needs to change, then please update `nrf54h20dk_nrf54h20_cpuapp.yaml` as well.

File: boards/nordic/nrf54h20dk/nrf54h20dk_nrf54h20-memory_map.dtsi [hunk 3]
```diff
@@ -174,11 +174,26 @@
...
+		compatible = "nordic,owned-partitions", "fixed-partitions";
+		status = "disabled";
+		perm-read;
+		perm-write;
+		perm-secure;
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		storage_partition: partition@130000 {
+			label = "storage";
```
Comment (57300) [inline boards/nordic/nrf54h20dk/nrf54h20dk_nrf54h20-memory_map.dtsi:195]: I don't think these labels are used anymore.

**Review by 57300 — CHANGES_REQUESTED** [review]

File: boards/nordic/nrf54h20dk/nrf54h20dk_nrf54h20-memory_map.dtsi (hunk 2 above)
Comment (hubertmis) [reply boards/nordic/nrf54h20dk/nrf54h20dk_nrf54h20-memory_map.dtsi:177]: If my calculations are correct, the `cpuapp_rw_partitions` is placed 552 KB further than this partition. So there is no need to shrink it. It could be even extended to 552 KB if needed.

BTW, is there a reason for this 64kB hole between the regions?

**Comment by ppryga-nordic** [comment]
I'm wondering why compliance doesn't see KConfig symbols defined in Zephyr...

**Comment by 57300** [comment]
> I'm wondering why compliance doesn't see KConfig symbols defined in Zephyr...

It's because `Kconfig.<board_name>` is part of the reusable board/SoC Kconfig subtree, introduced with HWMv2. This subtree is reused by sysbuild, so it must not contain settings that are undefined in sysbuild, and compliance can now check for this.

File: boards/nordic/nrf54h20dk/Kconfig.nrf54h20dk (hunk 1 above)
Comment (57300) [reply boards/nordic/nrf54h20dk/Kconfig.nrf54h20dk:7]: Now revert all changes to `Kconfig.nrf54h20dk`.

**Comment by ppryga-nordic** [comment]
@57300 requested change done. Fixed CI and compliance. @anangl  thank you for your help.

If there are no more objections then PR can be merged and settings should be working.

**Comment by ppryga-nordic** [comment]
Added a commit with nRF54H20 board configuration for hci_ipc BLE sample.

**Review by hubertmis — DISMISSED** [review]

**Review by 57300 — DISMISSED** [review]

File: boards/nordic/nrf54h20dk/nrf54h20dk_nrf54h20_cpuapp.yaml [hunk 4]
```diff
@@ -10,6 +10,6 @@ toolchain:
   - xtools
   - zephyr
 ram: 256
-flash: 512
+flash: 600
```
Comment (anangl) [inline boards/nordic/nrf54h20dk/nrf54h20dk_nrf54h20_cpuapp.yaml:13]: Why is this modified? `cpuapp_slot0_partition` size is still 512 kB.

Comment (ppryga-nordic) [reply boards/nordic/nrf54h20dk/nrf54h20dk_nrf54h20_cpuapp.yaml:13]: Isn't it size of flash used ? @57300 asked to update it.

Comment (anangl) [reply boards/nordic/nrf54h20dk/nrf54h20dk_nrf54h20_cpuapp.yaml:13]: No, it's the available flash. See [Board configuration](https://docs.zephyrproject.org/latest/develop/test/twister.html#board-configuration).

Comment (ppryga-nordic) [reply boards/nordic/nrf54h20dk/nrf54h20dk_nrf54h20_cpuapp.yaml:13]: Ok then I'll revert it

**Comment by ppryga-nordic** [comment]
@anangl change reverted.

File: boards/nordic/nrf54h20dk/nrf54h20dk_nrf54h20_cpuapp.yaml (hunk 4 above)
Comment (ppryga-nordic) [reply boards/nordic/nrf54h20dk/nrf54h20dk_nrf54h20_cpuapp.yaml]: done

**Review by anangl — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Review by 57300 — APPROVED** [review]

**Comment by ppryga-nordic** [comment]
@carlescufi  @nashif could on of you merge the PR?
